compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1417,7 +1281,8 @@ private boolean isKeyword(ParserRuleContext currentCtx) {
         }
     }
 
-    private boolean hasAlternativePaths(ParserRuleContext currentCtx) {
+    @Override
+    protected boolean isProductionWithAlternatives(ParserRuleContext currentCtx) {
```

Comment:
shall keep `hasAlternativePaths`? It is short and gives the required meaning.  We have a separate category for "Production" in ParserRuleContext.java but we reach here not only for them.